src/mesh/coordinates.cxx

```diff
@@ -313,13 +313,13 @@ Coordinates::Coordinates(Mesh* mesh, Options* options)
       output_warn.write("Not all covariant components of metric tensor found. "
                         "Calculating all from the contravariant tensor\n");
       /// Calculate contravariant metric components if not found
-      if (calcCovariant()) {
+      if (calcCovariant("RGN_NOBNDRY")) {
```

Comment:
I think we want `RGN_NOCORNERS` from #1719 here: `interpolateAndExtrapolate` won't always fill the boundary cells - it doesn't extrapolate if the boundary cells were read from the grid file - so then we need to call invert3x3 in the boundary cells (but not the corners).